src/raster/r.suitability.regions/r.suitability.regions.py

```diff
@@ -0,0 +1,410 @@
+#% guisection: Input
+#%end
+#%option
+#% key: absminsuit
+#% description: Threshold below which cells are marked as fully unsuitable
+#% required: no
+#% type: string
+#% key_desc: float
+#% guisection: Input
+#%end
+
+#%flag
+#% key: d
+#% label: Clumps including diagonal neighbors
+#% description: Clumps including diagonal neighbors
+#% guisection: Optional
+#%end
+
+#%option
+#% key: radius
+#% description: Radius for focal stat
+#% required: no
+#% type: integer
+#% answer: 1
+#% guisection: Focal stats
+#%end
+#%flag
+#% key: c
+#% label: Circular neighborhood
+#% description: Use circular neighborhood for focal statistic
+#% guisection: Focal stats
+#%end
+#%option
+#% key: focalstat
+#% description: Neighborhood operation (focal statistic)
+#% required: no
+#% type: string
+#% answer: median
+#% options:maximum,median,quart1,quart3
+#% guisection: Focal stats
+#%end
+
+#%option
+#% key: maxgap
+#% description: Maximum gap size (in hectares) to remove
+#% required: no
+#% type: string
+#% answer: 0
+#% key_desc: float
+#% guisection: Remove gaps
+#%end
+
+#%flag
+#% key: z
+#% label: Suitability per region
+#% description: Compute map with average suitability per region
+#% guisection: Reporting stats
+#%end
+#%flag
+#% key: a
+#% label: Calculates area of clumped areas (hectares)
+#% description: Calculates area of clumped areas
+#% guisection: Reporting stats
+#%end
+
+#%flag
+#% key: k
+#% label: Keep map with all suitable cells
+#% description: Keep map that show suitable areas (irrespective of clump size)
+#% guisection: Optional
+#%end
+
+#%flag
+#% key: f
+#% label: Keep suitability based on focal statistics
+#% description: Keep suitabiliyt based on focal statistics
+#% guisection: Optional
+#%end
+
+import sys
+import os
+import atexit
+import uuid
+import grass.script as gs
+
+RECLASSRULES1 = """\
+1:filled gaps
+2:suitable areas
+"""
+COLORRULES1 = """
+1 230:230:230
+""".strip()
+
+COLORRULES2 = """
+1 241:241:114
+2 139:205:85
+""".strip()
+
+
+CLEAN_LAY = []
+
+
+def cleanup():
+    """Remove temporary maps specified in the global list"""
+    cleanrast = list(reversed(CLEAN_LAY))
+    for rast in cleanrast:
+        ffile = gs.find_file(name=rast, element="cell", mapset=gs.gisenv()["MAPSET"])
+        if ffile["file"]:
+            gs.run_command("g.remove", flags="f", type="raster", name=rast, quiet=True)
+
+
+def tmpname(prefix):
+    """Generate a tmp name which contains prefix
+    Store the name in the global list.
+    Use only for raster maps.
+    """
+    tmpf = prefix + str(uuid.uuid4())
+    tmpf = tmpf.replace("-", "_")
+    CLEAN_LAY.append(tmpf)
+    return tmpf
+
+
+def tmpmask(input, absmin):
+    """Create tmp mask"""
+    rules = "*:{}:1".format(absmin)
+    tmprecode = tmpname("tmprecode")
+    gs.write_command(
+        "r.recode", input=input, output=tmprecode, rule="-", stdin=rules, quiet=True
+    )
+    return tmprecode
+
+
+def main(options, flags):
+
+    suitrast = options["suitrast"]
```

Comment:
Similarly to GRASS option names, the current trend in Python is using real words plus some special cases such as num, x, etc. for variable names and underscores to separate words.

Reply:
Implemented. But ambiguous feelings about this, as it is difficult to find names that make sense and still are not too long.